### ISSUE TITLE
Runtime of linear mixture models

### ISSUE THREAD

**Issue by yallup** [handley-lab/lsbi#32]
**Describe the bug**
rvs(N) call on a linear mixture model scales badly with N

**To Reproduce**
```python
import numpy as np
from lsbi.model import LinearMixtureModel
import timeit


np.random.seed(0)

k, d, n = 10, 100, 2
x = np.arange(d)
m = np.random.randn(k, d) * 3
mu = np.random.randn(k, n) * 3
M = np.random.randn(k, d, n)


C = np.eye(d) * 50
model = LinearMixtureModel(M=M, m=m, mu=mu, C=C)


def time_n(n):
    start_time = timeit.default_timer()
    model.evidence().rvs(n)
    end_time = timeit.default_timer()

    execution_time = end_time - start_time
    print(f"Execution time: {execution_time:2f} seconds, {n} samples")


for n in [1000, 10000, 100000 , 1000000]:
    time_n(n)

```

**Expected behavior**
linear runtime scaling with n, instead seeing:

python mixture_speed.py
Execution time: 0.337229 seconds, 1000 samples
Execution time: 0.675572 seconds, 10000 samples
Execution time: 1.635145 seconds, 100000 samples
...

and then nothing (appears to throttle at 1M samples), perhaps some hardware limit I will check

**Comment by yallup** [comment]
Just ran the above on my old laptop and it throttles at 100k samples, so I imagine there is some memory limitation in the operations being used

**Comment by williamjameshandley** [comment]
To what extent is this resolved by #34?

**Comment by yallup** [comment]
Largely resolved
